core-api/src/telemetry/metrics.rs

```diff
@@ -132,6 +317,11 @@ pub trait CustomMetricAttributes: Debug + Send + Sync {
     /// Must be implemented to work around existing type system restrictions, see
     /// [here](https://internals.rust-lang.org/t/downcast-not-from-any-but-from-any-trait/16736/12)
     fn as_any(self: Arc<Self>) -> Arc<dyn Any + Send + Sync>;
+
+    /// Return the stringified value for a label key, if available.
+    fn label_value(&self, _key: &str) -> Option<String> {
+        None
+    }
```

Comment:
I'm not totally clear why we need this here or why the in memory meters need to implement this. I bring it up b/c I don't love having to add an impl of this to the C bridge, which shouldn't need to do anything different because we're now recording in-mem metrics (same for metrics buffer)